code/obj/critter/misc.dm

```diff
@@ -487,22 +491,45 @@
 			else
 				continue
 
+	attackby(obj/item/W as obj, mob/M as mob)
+		if(istype(W, /obj/item/reagent_containers/food/snacks) && !(M in src.friends))
```

Comment:
If you want healing use:
`src.health += src.health_gain_from_food` as the parent does it.
I also recommend adding `eat_twitch(src)` for visual neatness.